frontend/app/login/page.tsx

```diff
@@ -1,7 +1,165 @@
-"use-client"
-import React from "react"
+"use client"
+import React, { useState } from "react"
+import Typography from "@mui/material/Typography"
+import Grid from "@mui/material/Grid"
+import TextField from "@mui/material/TextField"
+import Button from "@mui/material/Button"
+import { createTheme, ThemeProvider } from "@mui/material/styles"
+import { LoginValidation } from "@/app/helpers/login.helper"
 
-const Page = (): React.ReactElement => {
-  return <h1>Login Page</h1>
+const Login = (): React.ReactElement => {
+  const [credentials, setCredentials] =
+    useState<UserCredentials>(defaultCredential)
+  const [validationError, setValidationError] = useState<ValidationError>(
+    defaultValidationError
+  )
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = event.target
+    setCredentials({
+      ...credentials,
+      [name]: value,
+    })
+  }
+
+  const onLogin = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    event.preventDefault()
+    LoginValidation(credentials, setValidationError)
+  }
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        height: "100vh",
+      }}
+    >
+      <Grid sx={{ maxWidth: 545, p: 5 }}>
```

Comment:
what is the p for?

Reply:
The p is shorthand for the padding property in MUI. But p: 5 here 5 does not mean 5 pixels but  is 40px based on their internal calculations of MUI, however if we want to define that Grid for 5px only then we can use padding: "5px" then p